### ISSUE TITLE
Upgraded to Version 0.2.3-beta and lost motion binary sensors

### ISSUE THREAD

**Issue by pdobrien3** [uvjustin/alarmdotcom#106]
just upgraded to the beta. everything went fine.  the new config is nice. motion sensor binary sensors and their battery sensors were missing though

**Comment by elahd** [comment]
@pdobrien3  I deliberately blocked those in this release. The integration doesn't get instant feedback from Alarm.com -- by default, it only checks for updates once a minute -- so I didn't think that anyone would be able to use them reliably. Do you use motion sensors in automations? Happy to add them back in.

**Comment by pdobrien3** [comment]
No. That makes perfect sense. I added the motions just because they were there. While the actual binary sensors were not beneficial, the malfunction and battery ones were a welcome addition. Would it make sense to just add those back?

**Comment by elahd** [comment]
Interesting, that's a good idea. That'll take a bit more work, though, than adding it all back, so I'll keep this open for a future release. Thanks for the feedback!

**Comment by pdobrien3** [comment]
thank you for what you are doing. also, i have a SMKTX-345 smoke detector. the high heat and smoke seem to properly report.  the battery and malfunction seem to work fine for the freeze alarm but binary_sensor.freeze_alarm reports as unknown. in prior releases it reported malfunction i think. should i make another issue or this is good?

**Comment by elahd** [comment]
@Valdorama My concern is that Home Assistant will never see an motion event and will therefore always mark a room as unoccupied. As an example, if a motion sensor enters a "detected" state for 5 seconds after being triggered and this integration only polls for updates once every 60 seconds, odds are that you'll never receive the trigger event. Can you tell me for how long a motion sensor reports as active after being tripped?

@pdobrien3 What's your level of technical skill? Are you comfortable downloading a Python library using pip and using the command line?

**Comment by AritroSaha10** [comment]
> Can you tell me for how long a motion sensor reports as active after being tripped?

 @elahd The motion sensors generally stay active for a while. In my experience, it seems to stay on for at least 30 minutes.

**Comment by elahd** [comment]
@AritroSaha10 @Valdorama Thanks. Back in it goes. I'll push another beta update later today.

@pdobrien3 No need to mess with your Home Assistant server. You can do this on your laptop/desktop.

1. [Install python>3.9](https://www.python.org/downloads/).
2. Open a command prompt and [install pip](https://pip.pypa.io/en/stable/installation/) by entering `python -m ensurepip --upgrade` on Mac or `py -m ensurepip --upgrade` on Windows. 
3. Install pyalarmdotcomajax (the library that powers this integration) via pip: `pip install pyalarmdotcomajax`
4. Run `adc -u YOUR_USERNAME -p YOUR_PASSWORD -v`. Use quotation marks around your password if it contains special characters. Enter your OTP when prompted.
5. The CLI will return a dump of raw data organized by device type. Copy the entire JSON object in the `====[SENSORS]====` section into a JSON formatter, like [jsonformatter.org](https://jsonformatter.org/).
6. Search for the freeze sensor object in the formatted output. (Just Ctrl+F for whatever you named the sensor on Alarm.com.) Post the sensor's JSON object here, but be sure to redact the sensor `id` field, any `id` fields in the `relationships` object, and the sensor's `description` field, if you care. You should be posting something that looks like this:

```json
    {
      "id": "XXXXX",
      "type": "devices/sensor",
      "attributes": {
        "deviceType": 5,
        "openClosedStatus": 2,
        "state": 1,
        "isFlexIO": false,
        "deviceRole": 0,
        "displayStateText": "Not Reset",
        "isMonitoringEnabled": false,
        "sensorNamingFormat": 3,
        "managedDeviceType": 2,
        "hasState": false,
        "canBeRenamed": true,
        "canBeDeleted": false,
        "canAccessWebSettings": true,
        "canAccessAppSettings": false,
        "webSettings": 400,
        "canAccessTroubleshootingWizard": false,
        "troubleshootingWizard": null,
        "macAddress": "",
        "manufacturer": "",
        "isOAuth": false,
        "isZWave": false,
        "supportsCommandClassBasic": false,
        "isMalfunctioning": false,
        "canBeSaved": true,
        "canChangeDescription": true,
        "description": "Interconnected Smoke Detectors",
        "deviceModelId": 114,
        "canConfirmStateChange": true,
        "canReceiveCommands": false,
        "remoteCommandsEnabled": true,
        "hasPermissionToChangeState": true,
        "deviceIcon": {
          "icon": 218
        },
        "batteryLevelNull": null,
        "lowBattery": false,
        "criticalBattery": false,
        "canBeLinkedToVideoDevice": false,
        "videoDeviceLinkDeviceIds": []
      },
      "relationships": {
        "system": {
          "data": {
            "id": "XXXXX",
            "type": "systems/system"
          }
        },
        "stateInfo": {
          "data": null
        }
      }
    },
```

Thanks!

**Comment by pdobrien3** [comment]
Looks easy enough. Hopefully I can take a shot at it on Friday or Saturday. Thanks.

**Comment by elahd** [comment]
Motion sensors restored in v0.2.3-beta.2.

**Comment by pdobrien3** [comment]
So I worked through the steps this am.  A couple questions, if you don't mind, from a beginner.  Do you typically run the python commands from the python program itself, a command prompt, or secure shell?  Do you some how create a venv (running in windows) or just bring up a prompt and start working?

Also, I noticed this for my freeze part of the smoke detector:
   "isMonitoringEnabled": false,

I logged into alarm.com and noticed that all three parts of the smoke detector, along with the firefighter, show up as N/A under device/monitoring but the only sensor that doesn't look like it is working is the freeze alarm from the smoke detector. When i get a minute I need to check "isMonitoringEnabled" on the other three sensors.

Also, I upgraded to beta 2 and all motions are back.

**Comment by pdobrien3** [comment]
{
      "id":,
      "type": "devices/sensor",
      "attributes": {
        "deviceType": 8,
        "openClosedStatus": 0,
        "state": 0,
        "isFlexIO": false,
        "deviceRole": 0,
        "displayStateText": "Status Unknown",
        "isMonitoringEnabled": false,
        "sensorNamingFormat": 3,
        "managedDeviceType": 2,
        "hasState": true,
        "canBeRenamed": true,
        "canBeDeleted": false,
        "canAccessWebSettings": true,
        "canAccessAppSettings": false,
        "webSettings": 400,
        "canAccessTroubleshootingWizard": false,
        "troubleshootingWizard": null,
        "macAddress": "",
        "manufacturer": "",
        "isOAuth": false,
        "isZWave": false,
        "supportsCommandClassBasic": false,
        "isMalfunctioning": false,
        "canBeSaved": true,
        "canChangeDescription": true,
        "description":
        "deviceModelId": 117,
        "canConfirmStateChange": true,
        "canReceiveCommands": false,
        "remoteCommandsEnabled": true,
        "hasPermissionToChangeState": true,
        "deviceIcon": {
          "icon": 317
        },
        "batteryLevelNull": null,
        "lowBattery": false,
        "criticalBattery": false,
        "canBeLinkedToVideoDevice": false,
        "videoDeviceLinkDeviceIds": []
      },
      "relationships": {
        "system": {
          "data": {
            "id": ,
            "type":
          }
        },
        "stateInfo": {
          "data": null
        }
      }
    },

**Comment by elahd** [comment]
Thanks, @pdobrien3. Take a look at my answer to another issue [here](https://github.com/uvjustin/alarmdotcom/issues/45#issuecomment-1018880202). It may be possible that you don't have monitoring enabled for these sensors in the UI.

Also, would you mind updating to v0.2.3-beta.3 to test freeze sensors?

**Comment by pdobrien3** [comment]
Right, but where the checkboxes is under devices, says N/A. I have gotten a response from my monitoring company. They said those devices are security devices  bs sensor devices

`That section in Alarm.com is referencing sensor activity monitoring, not alarm monitoring. Life safety devices are either in alarm status or at rest, they are not eligible for sensor activity monitoring like door window sensors or motion detectors.`

off to check beta 3 now

**Comment by elahd** [comment]
That's weird. This may be specific to your provider. `"state": 0` maps to "unknown." Does the sensor at least show up in Home Assistant as a `cold` sensor in beta 3?x

I don't have any freeze sensors, but my CO and smoke detectors return state (see what I pasted in an earlier post).

They show up on the ADC website as:

<img width="1023" alt="image" src="https://user-images.githubusercontent.com/466460/162354365-6b0c91df-1b24-4384-ab94-f2c239ee4189.png">

**Comment by pdobrien3** [comment]
Yes, it has always shown up as a freeze alarm.  I think the icon changed.  What is odd to me is that the smoke, high heat, and freeze are all one device.  The smoke and high heat seem to have a state but the freeze does not. 

Have you seen if any of your alarm monitoring devices will actually change state?  According to my provider, anything with an N/A under monitoring is not meant to be monitored like a sensor although they did say battery and malfunction state was monitored. I have a firefighter that is listed as N/A. I may see if I can get the state on it to change this weekend.

**Comment by pdobrien3** [comment]
Also, I was seeing multiple adc login attempts to update errors that have vanished. Not sure it that is something with the recent beta or just something on my network that has improved. 

Also, I have been on a 15 sec scan interval for some time now without any issues.

**Comment by elahd** [comment]
> Have you seen if any of your alarm monitoring devices will actually change state?

I haven't actually tested this. I can try next week. Let me know what you find on your end.

> Also, I was seeing multiple adc login attempts to update errors that have vanished. 

That's great. Were you seeing the errors on the ADC website or in Home Assistant?

**Comment by pdobrien3** [comment]
Just a heads up.  I was planning on testing some of the security sensors today.  Before I did, I tested scan interval.  It didnt seem to make a difference if it was set to 15, 30 or 60 seconds.  I had response times from 45 seconds all the way to 1:45.  the 45 seconds was obtained when scan interval was set to 60 seconds.

**Comment by elahd** [comment]
Thanks! Fixed in v0.2.5.
